File: node-js-test/src/domain/repositories/company.repository.ts
import { BaseRepository } from 'src/shared/bases/base.repository';
import { CompanyEntity } from '../entities/company.entity';

export abstract class CompanyRepository extends BaseRepository<CompanyEntity> {
  abstract findByCnpj(cnpj: string): Promise<CompanyEntity | null>;
  abstract findByEmail(email: string): Promise<CompanyEntity | null>;
}

File: golang-test/internal/domain/repository/company_repository.go
package repository

import (
	"context"

	"github.com/mateh3007/karhub/golang-test/internal/domain/entity"
)

type CompanyRepository interface {
	Create(ctx context.Context, company *entity.Company) (*entity.Company, error)
	FindByID(ctx context.Context, id string) (*entity.Company, error)
	Update(ctx context.Context, company *entity.Company) (*entity.Company, error)
	Delete(ctx context.Context, id string) error
	FindByCNPJ(ctx context.Context, cnpj string) (*entity.Company, error)
	FindByEmail(ctx context.Context, email string) (*entity.Company, error)
}
